uniffi_bindgen/src/bindings/kotlin/gen_kotlin/mod.rs

```diff
@@ -99,20 +99,19 @@ pub struct CustomTypeConfig {
 }
 
 impl Config {
+    // We insist someone has already configured us - any defaults we supply would be wrong.
```

Comment:
Slightly confused by this one.  Sometimes we do use the default package name and it's okay right?

I'm guessing that this function is only called in some specific circumstances where we do need an explicit package name, like maybe for external types?  If so, how about renaming the function to reflect that?

Reply:
> Slightly confused by this one. Sometimes we do use the default package name and it's okay right?

No - the problem is that there are 2 places a default is applied - eg, for kotlin, [here](https://github.com/mozilla/uniffi-rs/blob/962de986f02f726954b1dec2f5014d8f99828439/uniffi_bindgen/src/bindings/kotlin/gen_kotlin/mod.rs#L106) and by the binding generator [here](https://github.com/mozilla/uniffi-rs/blob/962de986f02f726954b1dec2f5014d8f99828439/uniffi_bindgen/src/bindings/kotlin/mod.rs#L38) - and as you can see, two different default values exist.

So we do use a default package name, but never use *this* default package name.

> I'm guessing that this function is only called in some specific circumstances where we do need an explicit package name,

The function is used by the templates - it's just never used when `self.package_name` remains None.

Reply:
I'm starting to understand this one.  I think the `expect` message say something like `package name should have been set in update_component_configs`.  The way it currently reads, it seems like the issue was that the user didn't set it in the `config.toml`, but IIUC the issue is actually in the code.

Reply:
> The way it currently reads, it seems like the issue was that the user didn't set it in the config.toml, but IIUC the issue is actually in the code.

Yeah, that's a very good point, I'll update the message.